### DISCUSSION TITLE
Logging: compiled vs default module log level

### DISCUSSION THREAD

**Discussion by ottojo** [zephyrproject-rtos/zephyr#36224] [Q&A]
Hi!
I know you can set the maximum *compiled* logging level in a module using
```
LOG_MODULE_REGISTER(controller, LOG_LEVEL_INF);
```

Is there a way to specify the default logging level at startup? I would like to have the logging level at WRN by default, but be able to change to DBG at runtime (ideally per module).

**Comment by yashi** [comment]
There are a few `CONFIG_LOG_*` for that.

- `LOG_RUNTIME_FILTERING`
- `LOG_DEFAULT_LEVEL`
- `LOG_MAX_LEVEL`

For your case, set `CONFIG_LOG_DEFAULT_LEVEL` to 2, `LOG_MAX_LEVEL` to 4, and call `log_filter_set()` with `LOG_LEVEL_DBG`.  You can pass `source_id` to select a specific module.

**Reply by lnhao** [reply]
Hi @yashi, when my board is using the RTT interface to print output, which is the first parameter `backend `pointer used in the `log_filter_set()` function?

**Comment by gschwaer** [comment]
This doesn't work with the shell command `kernel log-level module_name 4`, does it? At least it doesn't work for me sadly.
